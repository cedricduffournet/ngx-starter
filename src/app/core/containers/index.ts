export * from './toaster';
export * from './http-progress-bar/http-progress-bar.component';
