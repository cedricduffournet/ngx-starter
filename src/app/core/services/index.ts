export * from './http-progress.interceptor';
export * from './http.service';
