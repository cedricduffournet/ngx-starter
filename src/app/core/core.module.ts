import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { environment } from '@env/environment';
import {
  HttpClientModule,
  HttpClient,
  HTTP_INTERCEPTORS
} from '@angular/common/http';

import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { ModalModule } from 'ngx-bootstrap/modal';
import { ToasterModule } from 'angular2-toaster';

import {
  ToasterComponent,
  HttpProgressBarComponent
} from '@app/core/containers';
import { HttpProgressInterceptor } from '@app/core/services/http-progress.interceptor';

const MODULES: any[] = [
  HttpClientModule,
  TranslateModule.forRoot({
    loader: {
      provide: TranslateLoader,
      useFactory: HttpLoaderFactory,
      deps: [HttpClient]
    }
  }),
  ModalModule.forRoot(),
  ToasterModule.forRoot(),
  CommonModule
];

export const COMPONENTS = [ToasterComponent, HttpProgressBarComponent];

@NgModule({
  imports: MODULES,
  declarations: COMPONENTS,
  exports: COMPONENTS,
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: HttpProgressInterceptor,
      multi: true
    }
  ]
})
export class CoreModule {
  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule
  ) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import only once in main AppModule.'
      );
    }
  }
}

export function HttpLoaderFactory(http: HttpClient) {
  return new TranslateHttpLoader(
    http,
    `${environment.prefixAsset}/assets/i18n/`,
    '.json'
  );
}
