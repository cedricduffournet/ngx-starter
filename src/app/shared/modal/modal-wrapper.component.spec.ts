import { TestBed, ComponentFixture } from '@angular/core/testing';

import { ModalWrapperComponent } from '@app/shared/modal/modal-wrapper.component';

describe('ModalWrapperComponent', () => {
  let fixture: ComponentFixture<ModalWrapperComponent>;
  let component: ModalWrapperComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [ModalWrapperComponent]
    });
    fixture = TestBed.createComponent(ModalWrapperComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should be created with title', () => {
    component.title = 'test title';

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
