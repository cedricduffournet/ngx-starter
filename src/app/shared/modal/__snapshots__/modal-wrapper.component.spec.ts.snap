// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapperComponent should be created 1`] = `
<app-modal-wrapper
  title=""
>
  <div
    class="modal-content bg-light"
  >
    
    <div
      class="modal-body p-0"
    />
  </div>
</app-modal-wrapper>
`;

exports[`ModalWrapperComponent should be created with title 1`] = `
<app-modal-wrapper
  title={[Function String]}
>
  <div
    class="modal-content bg-light"
  >
    
    <div
      class="modal-header"
    >
      <h4
        class="modal-title"
      >
        test title
      </h4>
    </div>
    <div
      class="modal-body p-0"
    />
  </div>
</app-modal-wrapper>
`;
